### PR TITLE
Retrieve supported workspace classes  from workspace cluster

### PR DESCRIPTION
**Pull request by Furisto**
## Description
**Note:** This is a stacked PR that depends on https://github.com/gitpod-io/gitpod/pull/10982

During registration of a workspace cluster the workspace classes that are supported by this cluster will be fetched from the DescribeCluster endpoint of ws-manager. They are also fetched in a configurable interval (currently 60 seconds) so that updates to the supported classes are propagated to the meta side. 

I have added a feature flag, so that this change can be deployed even if the new endpoint from https://github.com/gitpod-io/gitpod/pull/10982 has not been deployed to workspace clusters yet.

## Related Issue(s)
Partially fixes #https://github.com/gitpod-io/gitpod/issues/10842

## How to test
- Currently the feature flag (workspace_classes_backend) is active, so with debug logs activated (gpctl debug logs ws-manager-bridge) you should see messages like `reconciling workspace classes...`in the log output. After disabling the flag and restarting ws-manager-bridge you should not see any messages.

- Registration of a workspace cluster can be done with `gpctl clusters register --name ws-cluster1 --hint-cordoned --hint-govern --tls-path ./wsman-tls/ --url 34.77.51.162:8081 --kubeconfig ./meta-cluster`. After that in the admissionConstraints column of d_b_workspace_cluster you should see an entry similar to this:

```
[
   {
      "type":"has-class",
      "id":"default",
      "displayName":"default"
   },
   {
      "type":"has-class",
      "id":"gitpodio-internal-xl",
      "displayName":"XL"
   }
]
```

You can create a workspace cluster with workspace preview. I already have created a VM which you can use for registration. Contact me for the credentials.

## Release Notes

```release-note
None
```
## Werft options:

- [x] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@Furisto: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/gitpod-db/src/typeorm/workspace-cluster-db-impl.ts
Comment (Furisto) [inline components/gitpod-db/src/typeorm/workspace-cluster-db-impl.ts:46]: I added this in addition to findFiltered because findFiltered does not return the tls config which we need to authenticate against the ws-manager endpoint

Comment (geropl) [reply components/gitpod-db/src/typeorm/workspace-cluster-db-impl.ts:46]: Jep. We did so to reduce traffic (bc it turned out to be a perf problem). See [comment below](https://github.com/gitpod-io/gitpod/pull/11010#discussion_r911154433).

**Comment by sagor999** [comment]
Removed myself and workspace team from reviewers, since this PR does not contain any files that we are code owners of.

**Comment by geropl** [comment]
FYI: Thomas and I had a quick sync, aligned and discussed some minor changes. @Furisto Please ping for re-review! :pray:

**Comment by Furisto** [comment]
@geropl ping

**Review by corneliusludmann — APPROVED** [review]
Changes on Team Self-Hosted owned files look good. Approving to unblock the merge after being reviewed by the other teams.

**Comment by geropl** [comment]
@Furisto I'm currently reviewing. Can you provide me the tls config for wsman? :pray:

**Comment by geropl** [comment]
@Furisto I received the credentials for your new VM. After setting everything up I still see no update of the admission constraints:
 1. ws cluster is configured and can connect
 2. bridge has run "reconcile workspace classes"
 3. feature flags are :green_circle:  in ConfigCat 
 4. still, the DB says no admission constraint:
 
image

Could it be that the ws cluster is missing the workspace classes config? :thinking:

**Review by geropl — APPROVED** [review]
Turns out the admissionsConstraint is properly updated which we can verify by accessing the DB directly. :heavy_check_mark: 

`gpctl` has to be taught to express workspaceClass admissionConstraints, but that's out-of-scope and worth a follow-up. :lotus_position:

**Pull request merged**
